refactor(language_server): Add formatter and linter features

leaysgur opened the pull request:
Currently, there is 2 ways to start LS for oxc apps.

- `oxc_language_server`
- `oxlint --lsp`
- (`oxfmt --lsp` will be added soon)


Having duplicates are fine for now, but `ServerFormatter` (and `oxc_formatter`) should not bundled in `oxlint --lsp` case.

This PR adds `linter` and `formatter` features to exclude.
For `oxc_language_server`, both features are enabled by default, keep it as-is.

Boshen commented:
We have https://github.com/oxc-project/oxc/issues/15738 down the road.

Sysix reviewed (approved):
oh here are the features flags :)

leaysgur committed:
refactor(language_server): Add formatter and linter features (#15767)

Currently, there is 2 ways to start LS for oxc apps.

- `oxc_language_server`
- `oxlint --lsp`
- (`oxfmt --lsp` will be added soon)

Having duplicates are fine for now, but `ServerFormatter` (and `oxc_formatter`) should not bundled in `oxlint --lsp` case.

This PR adds `linter` and `formatter` features to exclude.
For `oxc_language_server`, both features are enabled by default, keep it as-is.
